src/cosl/coordinated_workers/worker.py

```diff
@@ -406,6 +406,12 @@ def _add_readiness_check(self, new_layer: Layer):
 
     def _reconcile(self):
         """Run all unconditional logic."""
+        # There could be a race between the resource patch and pebble operations
+        # i.e., charm code proceeds beyond a can_connect guard, and then lightkube patches the statefulset
+        # and the workload is no longer available
+        if self.resources_patch and not self.resources_patch.is_ready():
```

Comment:
```suggestion
        if not self.resources_patch.is_ready():
```
or when could the `resources_patch` variable be falsy? please add a comment to explain

Reply:
`resources_patch` can be None when no resources are passed to the worker (don't set requests or limits). Added a comment